### DISCUSSION TITLE
sveltekit node-server X-Forwarded-Proto, X-Forwarded-Host

### DISCUSSION THREAD

**Discussion by yannishin** [sveltejs/kit#11900] [Q&A]
I expected the return from the Hook to be event.url.host=ui.foo.com, event.url.protocol=https:, but it is returning event.url.host=ui_server/ui, event.url.protocol=http:.

How should I solve this issue?

Given that event.request.headers.get("X-Forwarded-Host") and event.request.headers.get("X-Forwarded-Proto") are returning null, it seems that the node server(adapter-node.) is not recognizing the X-Forwarded-??? headers.

For reference, the API Server developed with spring boot is correctly returning ui.foo.com, https:.

This is Nginx Config

upstream ui_server {
  ip_hash;
 ui.foo.com:17001;
  keepalive 100;
}

    location /ui {

        proxy_pass  http://ui_server/ui;

        proxy_http_version 1.1;
        proxy_set_header Connection $connection_upgrade;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header        Host $host;
        proxy_set_header        X-Real-IP $remote_addr;
        proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header    X-Forwarded-Proto $scheme;
        proxy_set_header    X-Forwarded-Host $host;
        proxy_set_header Forwarded "for=$proxy_add_x_forwarded_for; host=$host; proto=$scheme";

        charset utf-8;

        access_log /opt/naon/app/nginx/logs/ui-access.log postdata;
      }

**Comment by teemingc** [comment]
Thanks for posting this. It sounds like a potential bug so I've created an issue for this here https://github.com/sveltejs/kit/issues/11902
